sampleParents.py

```diff
@@ -33,8 +33,13 @@ def sampleBestParents(fitnesses):
     fitnesses = np.array(fitnesses)
     prefixSums, totalSum = getPrefixSums(fitnesses)
 
-    samples = np.zeros((NUM_INDIVIDUALS,))
-    for i in range(NUM_INDIVIDUALS):
-        samples[i] = singleSample(prefixSums, totalSum)
+    samples = []
+    for _ in range(NUM_INDIVIDUALS):
+        parent1 = singleSample(prefixSums, totalSum)
+        parent2 = singleSample(prefixSums, totalSum)
+        while parent2 == parent1:
```

Comment:
Could this result in a infinite loop? Probably extremely unlikely but is there any other method for this?

Reply:
Yeah I also considered thinking longer for an alternate but wanted to make it quick for now. I think it's basically guaranteed to never hang, but you could do an unneccessary number of samples

Reply:
I think it runs fast enough right now where the inefficiency is alright unless you feel strongly that we should change it

Reply:
I think that's a fair assumption.